WordPress/Classes/ViewRelated/Media/MediaLibraryPickerDataSource.m

```
@@ -543,27 +557,38 @@ - (NSString *)name
     return NSLocalizedString(@"WordPress Media", @"Name for the WordPress Media Library");
 }
 
-- (WPMediaRequestID)imageWithSize:(CGSize)size completionHandler:(WPMediaImageBlock)completionHandler
+- (void)refreshImageMedia
 {
     NSString *entityName = NSStringFromClass([Media class]);
     NSFetchRequest *request = [NSFetchRequest fetchRequestWithEntityName:entityName];
     request.predicate = [MediaLibraryPickerDataSource predicateForFilter:self.filter blog:self.blog];
     NSSortDescriptor *sortDescriptor = [NSSortDescriptor sortDescriptorWithKey:@"creationDate" ascending:NO];
     request.sortDescriptors = @[sortDescriptor];
+    request.fetchLimit = 1;
     NSError *error;
     NSArray *mediaAssets = [[[ContextManager sharedInstance] mainContext] executeFetchRequest:request error:&error];
```

Sad that we swallow the error, but out of scope for this PR.

I wonder whether it's worth tracking an improvement task for this, or just wait till we'll get time to convert to Swift...